circus-train-aws-sns/src/main/java/com/hotels/bdp/circustrain/aws/sns/event/SnsListener.java

```diff
@@ -118,29 +118,44 @@ public void tableReplicationStart(EventTableReplication tableReplication, String
 
   @Override
   public void tableReplicationSuccess(EventTableReplication tableReplication, String eventId) {
-    String endTime = clock.getTime();
-    EventReplicaTable replicaTable = tableReplication.getReplicaTable();
-    SnsMessage message = new SnsMessage(SnsMessageType.SUCCESS, config.getHeaders(), startTime, endTime, eventId,
-        sourceCatalog.getName(), replicaCatalog.getName(), replicaCatalog.getHiveMetastoreUris(),
-        tableReplication.getSourceTable().getQualifiedName(), tableReplication.getQualifiedReplicaName(),
-        replicaTable.getTableLocation(), partitionKeyTypes,
-        getModifiedPartitions(partitionsToAlter, partitionsToCreate), getBytesReplicated(), null);
-    publish(config.getSuccessTopic(), message);
+    try {
+      String endTime = clock.getTime();
+      EventReplicaTable replicaTable = tableReplication.getReplicaTable();
+      SnsMessage message = new SnsMessage(SnsMessageType.SUCCESS, config.getHeaders(), startTime, endTime, eventId,
+          sourceCatalog.getName(), replicaCatalog.getName(), replicaCatalog.getHiveMetastoreUris(),
+          tableReplication.getSourceTable().getQualifiedName(), tableReplication.getQualifiedReplicaName(),
+          replicaTable.getTableLocation(), partitionKeyTypes,
+          getModifiedPartitions(partitionsToAlter, partitionsToCreate), getBytesReplicated(), null);
+      publish(config.getSuccessTopic(), message);
+    } finally {
+      // reset registered partitions between database cycles
+      partitionsToCreate = null;
+      partitionsToAlter = null;
+      partitionKeyTypes = null;
+    }
   }
 
   @Override
   public void tableReplicationFailure(EventTableReplication tableReplication, String eventId, Throwable t) {
-    if (startTime == null) {
-      startTime = clock.getTime();
+    try {
+      if (startTime == null) {
+
+        startTime = clock.getTime();
+      }
+      String endTime = clock.getTime();
+      EventReplicaTable replicaTable = tableReplication.getReplicaTable();
+      SnsMessage message = new SnsMessage(SnsMessageType.FAILURE, config.getHeaders(), startTime, endTime, eventId,
+          sourceCatalog.getName(), replicaCatalog.getName(), replicaCatalog.getHiveMetastoreUris(),
+          tableReplication.getSourceTable().getQualifiedName(), tableReplication.getQualifiedReplicaName(),
+          replicaTable.getTableLocation(), partitionKeyTypes,
+          getModifiedPartitions(partitionsToAlter, partitionsToCreate), getBytesReplicated(), t.getMessage());
+      publish(config.getFailTopic(), message);
+    } finally {
+      // reset registered partitions between database cycles
+      partitionsToCreate = null;
```

Comment:
I don't think this is covered by any tests?

Reply:
Done :)